Core/src/main/java/com/github/intellectualsites/plotsquared/plot/PlotSquared.java

```diff
@@ -1525,6 +1558,44 @@ public void copyFile(String file, String folder) {
     public void disable() {
         try {
             eventDispatcher.unregisterAll();
+            if (HybridUtils.UPDATE && Settings.Enabled_Components.PERSISTENT_ROAD_REGEN && (
+                !HybridUtils.regions.isEmpty() || !HybridUtils.chunks.isEmpty())) {
+                PlotSquared.log(Captions.PREFIX
+                    + "Road regeneration incomplete. Saving incomplete regions to disk.");
+                PlotSquared.debug("   Regions: " + HybridUtils.regions.size());
+                PlotSquared.debug("   Chunks: " + HybridUtils.chunks.size());
+                try {
+                    ArrayList<int[]> regions = new ArrayList<>();
+                    ArrayList<int[]> chunks = new ArrayList<>();
+                    for (BlockVector2 r : HybridUtils.regions) {
+                        regions.add(new int[] {r.getBlockX(), r.getBlockZ()});
+                    }
+                    for (BlockVector2 c : HybridUtils.chunks) {
+                        chunks.add(new int[] {c.getBlockX(), c.getBlockZ()});
+                    }
+                    List<Object> list = new ArrayList<>();
+                    list.add(regions);
+                    list.add(chunks);
+                    list.add(HybridUtils.height);
+                    File file = new File(
+                        this.IMP.getDirectory() + File.separator + "persistent_regen_data_"
+                            + HybridUtils.area.id + "_" + HybridUtils.area.worldname);
+                    if (file.exists() && !file.delete()) {
+                        PlotSquared.log(Captions.PREFIX
+                            + "persistent_regen_data file already exists and could not be deleted.");
+                    }
+                    if (!file.createNewFile()) {
+                        PlotSquared
+                            .log(Captions.PREFIX + "Error create persistent_regen_data file.");
+                    }
+                    ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(file));
+                    oos.writeObject(list);
+                    oos.flush();
+                    oos.close();
```

Comment:
Don't know why I didn't see that before, but `oos.close()` should be called in a `finally` block as it wouldn't close the resource on an exception otherwise. 

As an alternative way, you could use `Files.newOutputStream(Path.of(this.IMP.getDirectory(), "persistent_regen_data_" + HybridUtils.area.id + "_" + HybridUtils.area.worldname))` in a try-with-resource statement and therefore also minimize the scope of the try-catch block.

The same applies for reading obviously.

Reply:
Oof, idk why I put the closing inside the try in the first place...